src/ds/defines.h

```diff
@@ -20,6 +20,10 @@
 #  define SNMALLOC_COLD __attribute__((cold))
 #endif
 
+#ifdef __INTEL_COMPILER
+#  define SNMALLOC_ICPC_UNREACHABLE __builtin_unreachable()
+#endif
```

Comment:
This is a GCC builtin that's also supported by icc and clang.  It's probably a good idea to enable it for all compilers.  The MSVC spelling is `__assume(0)`.  The build is broken on all compilers in CI as a result of this.

Reply:
Oh, my fault. I forgot to add an empty definition for this. Yeah, adding this to all other compilers can be done but I PC needs this only because it is not being able to detect the fall through correctly across constexpr branches.